lib/ssl/test/ssl_mfl_SUITE.erl

```diff
@@ -0,0 +1,530 @@
+                         end
+                 end,
+    run_mfl_openssl(ErlProc, OpenSSL, MFL, Data, ErlRecvFun).
+
+run_mfl_openssl(ErlProc, OpenSSL, MFL, Data, ErlRecvFun) ->
+    MFL = get_openssl_max_fragment_length(OpenSSL),
+
+    true = port_command(OpenSSL, Data),
+    ErlRecvFun(),
+
+    ErlProc ! get_socket,
+    ErlSocket = receive
+                    {ErlProc, {socket, ErlSocket0}} ->
+                        ErlSocket0
+                end,
+    assert_mfl(ErlSocket, MFL),
+
+    RData = lists:reverse(Data),
+    flush(),
+    ssl:send(ErlSocket, RData),
+    RData = ssl_test_lib:active_recv(OpenSSL, length(RData)),
+    ok.
+
+%% ------------------------------------------------------------
+flush() ->
+    flush(false).
+flush(Noisy) ->
+    receive Rx ->
+            if Noisy ->
+                    io:format("~p:~p: ~999p~n", [self(), ?FUNCTION_NAME, Rx]);
+               true ->
+                    ignore
+            end,
+            flush(Noisy)
+    after 100 ->
+            ok
+    end.
+
+%% ------------------------------------------------------------
+get_openssl_max_fragment_length(Port) ->
+    get_openssl_max_fragment_length(Port, []).
+
+get_openssl_max_fragment_length(Port, Acc) ->
+    receive
+        {Port, {data, Data}} ->
+            get_openssl_max_fragment_length_line(Port, Acc++Data)
+    after 1000 ->
+            error(timeout)
+    end.
+
+get_openssl_max_fragment_length_line(Port, Acc) ->
+    case get_line(Acc) of
+        more ->
+            get_openssl_max_fragment_length(Port, Acc);
+        {"TLS "++TlsInfo, Acc2} ->
+            get_openssl_max_fragment_length_tlsinfo(TlsInfo, Port, Acc2);
+        {_Discard, Acc2} ->
+            get_openssl_max_fragment_length_line(Port, Acc2)
+    end.
+
+get_openssl_max_fragment_length_tlsinfo("client extension "++ExtInfo, Port, Acc) ->
+    get_openssl_max_fragment_length_ext(ExtInfo, Port, Acc);
+get_openssl_max_fragment_length_tlsinfo("server extension "++ExtInfo, Port, Acc) ->
+    get_openssl_max_fragment_length_ext(ExtInfo, Port, Acc);
+get_openssl_max_fragment_length_tlsinfo(_Acc, Port, Acc) ->
+    get_openssl_max_fragment_length_line(Port, Acc).
+
+get_openssl_max_fragment_length_ext("\"max fragment length\" (id=1), len=1"=Ext, Port, Acc) ->
+    case get_line(Acc) of
+        more ->
+            receive
+                {Port, {data, Data}} ->
+                    Acc1 = Acc++Data,
+                    get_openssl_max_fragment_length_ext(Ext, Port, Acc1)
+            after 1000 ->
+                    error(timeout)
+            end;
+        {"0000 - 01  "++_, _} ->
+            512;
+        {"0000 - 02  "++_, _} ->
+            1024;
+        {"0000 - 03  "++_, _} ->
+            2048;
+        {"0000 - 04  "++_, _} ->
+            4096
+    end;
+get_openssl_max_fragment_length_ext(_Acc, Port, Acc2) ->
+    get_openssl_max_fragment_length_line(Port, Acc2).
+
+
+get_line(Data) ->
+    get_line(Data, []).
+
+get_line([$\r|T], A) ->
```

Comment:
We will see in the nightly tests if we should handle old macOS line brakes, too.